src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,292 @@
+    private String password;
+    private String username;
+    private URL url;
+    private GsonBuilder gsonBuilder;
+    /** Defaults to {@link #MAX_CONNECTIONS} **/
+    private int maxConnections = MAX_CONNECTIONS;
+    private URL proxyURL;
+    private String proxyUser;
+    private String proxyPassword;
+    private boolean isSSLAuthenticationDisabled;
+    private SSLSocketFactory authenticatedModeSSLSocketFactory;
+    /** Defaults to {@link #CONNECTION_TIMEOUT} **/
+    private TimeoutOption connectionTimeout = DEFAULT_CONNECTION_TIMEOUT;
+    /** Defaults to {@link #READ_TIMEOUT} **/
+    private TimeoutOption readTimeout = DEFAULT_READ_TIMEOUT;
+
+
+    public static ClientBuilder account (String account) {
+        return new ClientBuilder(account);
+    }
+
+    public static ClientBuilder url(URL url) {
+        return new ClientBuilder(url);
+    }
+
+    public ClientBuilder(String account) {
+        //parse account into a url
+        assertNotEmpty(account, "accountName");
+        try {
+            this.url = URIBuilder.buildUri()
+                        .scheme("https")
+                        .host(account + ".cloudant.com")
+                        .port(443)
+                        .build().toURL();
+        } catch (MalformedURLException e) {
+            throw new CouchDbException(e);
+        }
+    }
+
+
+    public ClientBuilder(URL url) {
+        //Check if port exists
+        int port = url.getPort();
+        if (port < 0) {
+            port = url.getDefaultPort();
+        }
+        if (url.getUserInfo() != null) {
+            //Get username and password and replace credential variables
+            this.username = url.getUserInfo().substring(0, url
+                    .getUserInfo()
+                    .indexOf(":"));
+            this.password = url.getUserInfo().substring(url
+                    .getUserInfo()
+                    .indexOf(":"));
+        }
+
+        try {
+            //Remove user credentials from url
+            this.url = URIBuilder.buildUri()
+                    .scheme(url.getProtocol())
+                    .host(url.getHost())
+                    .port(port)
+                    .build().toURL();
+        } catch (MalformedURLException e) {
+            throw new CouchDbException(e);
+        }
+    }
+
+    public CloudantClient build() {
+
+        //Build properties and couchdb client
+        CouchDbProperties props = new CouchDbProperties(url.getProtocol(),
+                url.getHost(), url.getPort());
+
+        //Create cookie interceptor
+        if (this.username != null && this.password != null) {
+            //make interceptor if both username and password are not null
+
+            //Create cookie interceptor and set in HttpConnection interceptors
+            CookieInterceptor cookieInterceptor = new CookieInterceptor(username, password);
+
+            props.addRequestInterceptors(cookieInterceptor);
```

Comment:
Maybe just call `interceptors(cookieInterceptor)` here then we always get some coverage of the sub-class checking.